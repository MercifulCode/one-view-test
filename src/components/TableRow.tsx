import * as React from "react";

export interface ITableRowProps {
  cellContent: string[];
  onClick?(e: any): void;
}

export class TableRow extends React.Component<ITableRowProps> {
  public render() {
    const { cellContent = [], onClick = () => {} } = this.props;
    return (
      <tr onClick={onClick}>
        {cellContent.map((content, index) => (
          <td key={index}>{content}</td>
        ))}
      </tr>
    );
  }
}
