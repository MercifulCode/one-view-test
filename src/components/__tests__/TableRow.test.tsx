import { TableRow } from "../TableRow";
import { shallow } from "enzyme";

describe("TableRow", () => {
  it("Should render when given no data", () => {
    const wrapper = shallow(<TableRow cellContent={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render when given some data", () => {
    const wrapper = shallow(<TableRow cellContent={["One", "Two"]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
