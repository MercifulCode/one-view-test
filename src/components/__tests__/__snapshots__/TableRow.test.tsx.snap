// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow Should render when given no data 1`] = `
<tr
  onClick={[Function]}
/>
`;

exports[`TableRow Should render when given some data 1`] = `
<tr
  onClick={[Function]}
>
  <td
    key="0"
  >
    One
  </td>
  <td
    key="1"
  >
    Two
  </td>
</tr>
`;
